packages/data-context/test/unit/sources/RelevantRunsDataSource.spec.ts

```diff
@@ -147,5 +152,41 @@ describe('RelevantRunsDataSource', () => {
         commitsAhead: 0,
       })
     })
+
+    it('preserves running if switched', async () => {
```

Comment:
What does "switching" imply here - when you click on "Change to this run" (or whatever the button says)?

I'm guessing the reason no-one found this edge case in testing was because it was closely related to the timing of when you change runs?

Hope we can find a good way to E2E test this at some point - unit tests do the job, but I imagine it'd be hard to fully grok what this does from a user point of view without an E2E test.